functions/custom-post-types/post-type-events.php

```diff
@@ -0,0 +1,186 @@
+});
+
+add_filter('wpfc_fullcalendar_args', function($args) {
+    $args['meta_query'] = array(
+        array(
+            'key' => 'start_date_time',
+            'compare' => '>',
+            'value' => $_REQUEST['start'],
+        ),
+        array(
+            'key' => 'end_date_time',
+            'compare' => '<',
+            'value' => $_REQUEST['end'],
+        ),
+    );
+
+    return $args;
+});
+apply_filters('wpfc_fullcalendar_args', $args);
+
+
+add_filter('wpfc_ajax_post', function($item, $post) {
+    $start_date = get_field('start_date_time', $post->ID);
+    $end_date = get_field('end_date_time', $post->ID);
+
+    $start_date_dt = new DateTime($start_date);
+    $end_date_dt = new DateTime($end_date);
+
+    $item['start'] = date('Y-m-d\TH:i:s', $start_date_dt->getTimestamp());
+    $item['end']   = date('Y-m-d\TH:i:s', $end_date_dt->getTimestamp());
+
+    $color = "#aaa";
+
+    $colors = array();
+
+    foreach (get_the_terms($post->ID, 'event_category') as $term) {
+        $term_id = (int) $term->term_id;
+        $term_color = get_field('color', "event_category_{$term_id}");
+
+        if ($term_color) $colors[] = $term_color;
+    }
+
+    if (!count($colors)) {
+        // do nothing
+    } elseif (count($colors) === 1) {
+        $color = $colors[0];
+    } else {
+        $color = sprintf('linear-gradient(to left top, %s, %s)', $colors[0], $colors[1]);
+    }
+
+    $item['color'] = $color;
+
+    return $item;
+}, 10, 2);
+
+add_filter('wpfc_ajax', function($items) {
+    $new_items = array_filter($items, function($item) {
+        if (!isset($item['post_id'])) return false;
+
+        return user_should_access_event($item['post_id']);
+    });
+
+    return $new_items;
+});
+
+add_filter('wpfc_qtip_content', function($content) {
+    if (empty($_REQUEST['post_id'])) return $content;
+
+    $post_id = (int) $_REQUEST['post_id'];
+
+    $content .= get_field('description', $post_id);
+
+    return $content;
+});
+
+function event_is_public($event_id)
+{
+    return get_field('is_public', $event_id);
+}
+
+function event_allows_user_explicitly($event_id, $user_id)
+{
+    $allowed_users = get_field('allowed_users', $event_id);
+
+    if (!$allowed_users) return false;
+    if (!$user_id) return false;
+
+    return in_array($user_id, $allowed_users);
+}
+
+function user_should_access_event($event_id=null)
+{
+    if (is_null($event_id)) {
+        $event_id = get_the_ID();
+    }
+
+    if (event_is_public($event_id)) return true;
+    if (!is_user_logged_in()) return false;
+
+    $post = get_post($event_id);
+    if (!$post) return false;
+
+    $user_id = get_current_user_id();
+    if (current_user_can('administrator')) return true;
+    if (event_allows_user_explicitly($event_id, $user_id)) return true;
+    if ($user_id == $post->post_author) return true;
+
+    return false;
+}
+
+function jcake_event_guard() {
+  global $post;
+  if ( is_singular( 'events' ) && !user_should_access_event() ) {
```

Comment:
Would it be better to do it like this?
```suggestion
  if (!is_singular( 'events' )) return;
  if (user_should_access_event()) return;
```